opensrp-opd/src/test/java/org/smartregister/opd/utils/OpdJsonFormUtilsTest.java

```diff
@@ -0,0 +1,380 @@
+package org.smartregister.opd.utils;
+
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+
+import org.apache.commons.lang3.tuple.Triple;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import org.smartregister.Context;
+import org.smartregister.CoreLibrary;
+import org.smartregister.SyncConfiguration;
+import org.smartregister.clientandeventmodel.Event;
+import org.smartregister.location.helper.LocationHelper;
+import org.smartregister.opd.BuildConfig;
+import org.smartregister.opd.OpdLibrary;
+import org.smartregister.opd.configuration.OpdConfiguration;
+import org.smartregister.opd.configuration.OpdRegisterQueryProviderTest;
+import org.smartregister.opd.pojos.OpdMetadata;
+import org.smartregister.repository.AllSharedPreferences;
+import org.smartregister.repository.Repository;
+import org.smartregister.util.JsonFormUtils;
+
+import java.util.ArrayList;
+
+@PrepareForTest(OpdUtils.class)
+@RunWith(PowerMockRunner.class)
+public class OpdJsonFormUtilsTest {
+
+    @Rule
+    private ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void testGetFormAsJsonWithEmptyJsonObject() throws JSONException {
+        OpdMetadata opdMetadata = new OpdMetadata(OpdConstants.JSON_FORM_KEY.NAME
+                , OpdDbConstants.KEY.TABLE
+                , OpdConstants.EventType.OPD_REGISTRATION
+                , OpdConstants.EventType.UPDATE_OPD_REGISTRATION
+                , OpdConstants.CONFIG
+                , Class.class
+                , Class.class
+                , true);
+        OpdConfiguration opdConfiguration = new OpdConfiguration
+                .Builder(OpdRegisterQueryProviderTest.class)
+                .setOpdMetadata(opdMetadata)
+                .build();
+        OpdLibrary.init(PowerMockito.mock(Context.class), PowerMockito.mock(Repository.class), opdConfiguration,
+                BuildConfig.VERSION_CODE, 1);
+        expectedException.expect(JSONException.class);
```

Comment:
I brought this up before but I still don't see how checking for an exception thrown by an internal library helps test our own code?